### PR TITLE
typescript(vx-responsive): re-write package in TypeScript

### LINKED ISSUE #514: Re-write `@vx/vx-responsive` in TypeScript

### PR DESCRIPTION
**Pull request by Rudeg**
#### :rocket: Enhancements

This PR builds off of #488 which added TypeScript build support, and converts the `@vx/responsive` package to TypeScript.

Closes #514.

#### Testing
- [x] CI
- [x] Generates .d.ts files

@williaster @hshoff @schillerk @milesj @kristw

### CODE REVIEW

**Comment by Rudeg** [comment]
@williaster Thanks for the review! Should be good now :)

File: packages/vx-responsive/src/components/ParentSize.tsx [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+export default class ParentSize extends React.Component<
+  ParentSizeProps & JSX.IntrinsicElements['div'],
+  ParentSizeState
+> {
+  static defaultProps = {
+    debounceTime: 300,
+  };
+
+  animationFrameID: number | null;
+  ro: ResizeObserver | undefined;
```
Comment (kristw) [inline packages/vx-responsive/src/components/ParentSize.tsx]: do we want to use more verbose name?

File: packages/vx-responsive/src/enhancers/withParentSize.tsx [hunk 2]
```diff
@@ -1,73 +1,81 @@
...
+        parentWidth: width,
+        parentHeight: height,
+      });
+    }
+
+    render() {
+      const { parentWidth, parentHeight } = this.state;
+      return (
+        <div
+          style={{ width: '100%', height: '100%' }}
```
Comment (kristw) [inline packages/vx-responsive/src/enhancers/withParentSize.tsx]: Convert to constant. 
Note: This `100%` style is known to create a bug that leave some area of the screen empty when the children use less than 100% width or height, e.g. 50% width & 50% height.

File: packages/vx-responsive/src/enhancers/withScreenSize.tsx [hunk 3]
```diff
@@ -1,56 +1,62 @@
...
+    componentDidMount() {
+      window.addEventListener('resize', this.handleResize, false);
+      this.resize();
+    }
+
+    componentWillUnmount() {
+      window.removeEventListener('resize', this.handleResize, false);
+    }
+
+    resize(/** event */) {
```
Comment (kristw) [inline packages/vx-responsive/src/enhancers/withScreenSize.tsx]: use arrow function for binding?

File: packages/vx-responsive/src/enhancers/withParentSize.tsx [hunk 4]
```diff
@@ -1,73 +1,81 @@
...
+      });
+      if (this.container) this.ro.observe(this.container);
+    }
+
+    componentWillUnmount() {
+      if (this.animationFrameID) window.cancelAnimationFrame(this.animationFrameID);
+      if (this.ro) this.ro.disconnect();
+    }
+
+    resize({ width, height }: { width: number; height: number }) {
```
Comment (kristw) [inline packages/vx-responsive/src/enhancers/withParentSize.tsx]: use arrow function for binding?

**Comment by williaster** [comment]
looks like we might need to add types for `ResizeObserver` https://github.com/Microsoft/TypeScript/issues/28502

also, updated base branch to master.

**Comment by Rudeg** [comment]
@williaster I bumped `resize-observer-polyfill` to the `1.5.1` version, and it solved a problem with types. However, I'm getting lint error on `no-undef` for the type export. This rule produce false-positives for the TS instances like `types` and `interfaces`

**Comment by williaster** [comment]
Hrmm, that's strange 🙈 Do you think it'd be fixable by changing the syntax to something like

`export type ParentSizeProvidedProps = ParentSizeState;`

instead of 

export { ParentSizeState as ParentSizeProvidedProps }`?

**Comment by Rudeg** [comment]
@williaster that will work, thanks! Should be fixed now:)

File: packages/vx-responsive/src/components/ParentSize.tsx [hunk 5]
```diff
@@ -40,45 +41,39 @@ export default class ParentSize extends React.Component<
...
   render() {
     const { className, children, debounceTime, ...restProps } = this.props;
     return (
-      <div
-        style={{ width: '100%', height: '100%' }}
-        ref={this.setTarget}
-        className={className}
-        {...restProps}
-      >
+      <div style={PARENT_SIZE_STYLES} ref={this.setTarget} className={className} {...restProps}>
```
Comment (hshoff) [inline packages/vx-responsive/src/components/ParentSize.tsx]: could we make this a prop and default the prop value to PARENT_SIZE_STYLES?

**Pull request merged**